### PR TITLE
feat: implement discord-verification

### PR DESCRIPTION
**Pull request by HunteRoi**
Use @hunteroi/discord-verification to ask the user to link their Discord account with their student's email when joining the server.

### CODE REVIEW

**Comment by HunteRoi** [comment]
This new feature actually requires us to :
- add a `SENDGRID_API_KEY` entry in the .env file
- create a SendGrid account & generate an API key & a template to include them in the .env and configuration files
- create a `db.production.json` file on the server to save the users

I'd say this is still an early stage in the development of this feature. I would actually argue to merge it as is because we could work on several points such as:
- storing the data in an actual database instead of a JSON file
- using something else than SendGrid to send emails (or not)

**Review by DrakeXorn — DISMISSED** [review]

File: .gitignore [hunk 1]
```diff
@@ -56,11 +56,13 @@ typings/
...
+!.env.vault
 
 # next.js build output
 .next
 
-build
-/src/config.ts
\ No newline at end of file
+build/
+db.*.json
```
Comment (HunteRoi) [inline .gitignore]:
```suggestion
```

File: config.development.json [hunk 2]
```diff
@@ -104,5 +107,11 @@
         "emote": "📢"
     },
     "ok_hand": "👌",
-    "zeroWidthSpace": "​"
+    "zeroWidthSpace": "​",
+    "communicationServiceOptions": {
+        "mailData": {
+            "from": "tinael.devresse@henallux.be",
+            "templateId": "d-5e8429db595648749bcbcdf041d853ee"
```
Comment (HunteRoi) [inline config.development.json]:
```suggestion
            "from": "AN_EMAIL@DOMAIN.COM",
            "templateId": "TEMPLATE_ID"
```

File: src/config.ts [hunk 3]
```diff
@@ -27,7 +28,12 @@ const defaultConfig: Configuration = {
     tutor: { roleid: '', emote: '' },
     announce: { roleid: '', emote: '', channelid: '' },
     ok_hand: '',
-    zeroWidthSpace: ''
+    zeroWidthSpace: '',
+    communicationServiceOptions: {
+        apiKey: '',
+        mailData: { from: '', templateId: '' }
+    },
+    database: { fileName: '' }
```
Comment (HunteRoi) [inline src/config.ts]:
```suggestion
    }
```

File: src/models/Configuration.ts [hunk 4]
```diff
@@ -57,4 +59,10 @@ export interface Configuration {
 
     ok_hand: string;
     zeroWidthSpace: string;
+
+    communicationServiceOptions: SendGridOptions;
+
+    database: {
+        fileName: string;
+    };
```
Comment (HunteRoi) [inline src/models/Configuration.ts]:
```suggestion
```

**Comment by HunteRoi** [comment]
All that's left is to create a SendGrid account and a dynamic template ! 👯

File: README.md [hunk 5]
```diff
@@ -12,7 +12,6 @@ POSTGRES_USER=
 POSTGRES_PASSWORD=
 DATABASE_PORT=5432
 DATABASE_HOST=localhost
-DATABASE_URL=postgresql://${POSTGRES_USER}:${POSTGRES_PASSWORD}@${DATABASE_HOST}:${DATABASE_PORT}/${POSTGRES_DB}?schema=public
 NODE_ENV=development
```
Comment (HunteRoi) [inline README.md:15]:
```suggestion
NODE_ENV=development
SENDGRID_API_KEY=
```

File: config.development.json [hunk 6]
```diff
@@ -104,5 +107,11 @@
         "emote": "📢"
     },
     "ok_hand": "👌",
-    "zeroWidthSpace": "​"
+    "zeroWidthSpace": "​",
+    "communicationServiceOptions": {
+        "mailData": {
+            "from": "AN_EMAIL@DOMAIN.COM",
+            "templateId": "TEMPLATE_ID"
+        }
```
Comment (HunteRoi) [inline config.development.json]:
```suggestion
        "mailData": {
            "from": "mdpdevti@henallux.be",
            "templateId": "d-c8892944b4f2417bafece2d7a5b73d1f"
        }
```

File: config.production.json [hunk 7]
```diff
@@ -104,5 +107,11 @@
         "emote": "📢"
     },
     "ok_hand": "👌",
-    "zeroWidthSpace": "​"
+    "zeroWidthSpace": "​",
+    "communicationServiceOptions": {
+        "mailData": {
+            "from": "",
+            "templateId": ""
+        }
```
Comment (HunteRoi) [inline config.production.json]:
```suggestion
        "mailData": {
            "from": "mdpdevti@henallux.be",
            "templateId": "d-ae8dfdd7cc9c49f1af6e016ecbb4d856"
        }
```

File: package.json [hunk 8]
```diff
@@ -3,12 +3,11 @@
 	"version": "1.7.0",
```
Comment (HunteRoi) [inline package.json]:
```suggestion
	"version": "1.8.0",
```

File: README.md [hunk 9]
```diff
@@ -12,8 +12,8 @@ POSTGRES_USER=
...
+SENGRID_API_KEY=
```
Comment (HunteRoi) [inline README.md]:
```suggestion
SENDGRID_API_KEY=
```

**Review by DrakeXorn — APPROVED** [review]
Wow ! Nice job !

**Pull request merged**